### DISCUSSION TITLE
CSS files generated by Tailwind in several packages

### DISCUSSION THREAD

**Discussion by madiyetov** [tailwindlabs/tailwindcss#9807] [Help] [answered]
Let's say we have a UI library which generates css files with tailwind classes. And we have our frontend project which use that UI library and also uses tailwind to generate its own classes. In this case we will end up with two css files with the same class definitions. We can solve the conflicts by using `prefix` in UI library, but we will still have the same styles in both files. 

What do you do in this situation? Just load the same thing twice?

**Comment by pm0u** [comment, marked as answer]
We have a monorepo for this and although it's not awesome, I have the UI library generate it's own stylesheet (using rollup with `extract: true` so that it's a css file). This is used in our storybook for the components. I then have the web app scan the UI library components to generate classes in addition to it's own file structure. This, unfortunately, would not allow a prefix on the classes used in the UI library - I'm not sure if that's a requirement for you or just a workaround.

**Reply by pm0u** [reply]
Something like this:

```
const { colors } = require('../../configs/tailwind/brand-colors.js')
const path = require('path')

/**
 * Computes the path to the UI component library's component "containing directory"
 * This is dependent upon the entrypoint defined in the package.json for the UI components
 * and is a bit fragile.
 * 
 * @returns {string}
 */
const getUiComponentPath = () => {
  const main = require.resolve('package-name-for-ui-lib')
  const basename = path.basename(main)
  return path.relative(__dirname, main.replace(basename, ''))
}

/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    'components/**/*.{js,jsx,ts,tsx}',
    'pages/**/*.{js,jsx,ts,tsx}',
    `${getUiComponentPath()}/**/*.{js,jsx,ts,tsx}`
  ],
  theme: {
    colors
  },
  plugins: [],
}
```

```
// package.json in UI lib
{
 ...
   "main": "src"
 ...
}
 ```
 
 So, all the components are under the 'src' directory. the function basically gets the `src` directory based on the entry point defined in the package.json as `src/index.js`

**Reply by madiyetov** [reply]
Thank you for the answer 👍🏼
